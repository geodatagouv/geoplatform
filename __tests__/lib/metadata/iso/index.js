const {convert} = require('../../../../lib/metadata/iso')

describe('lib.metadata.iso.index', () => {
  describe('convert()', () => {
    it('extract featureTypes from a service metadata', () => {
      expect(convert({
        hierarchyLevel: 'service',
        identificationInfo: {
          serviceType: 'download',
          coupledResource: [
            {
              operationName: 'GetCapabilities',
              identifier: '9fd801e3-976a-49b6-b84a-0bd448fec8d5',
              scopedName: 'bati_avap_paimpol'
            },
            {
              operationName: 'DescribeFeatureType',
              identifier: '9fd801e3-976a-49b6-b84a-0bd448fec8d5',
              scopedName: 'bati_avap_paimpol'
            },
            {
              operationName: 'GetFeature',
              identifier: '9fd801e3-976a-49b6-b84a-0bd448fec8d5',
              scopedName: 'bati_avap_paimpol'
            },
            {
              operationName: 'GetCapabilities',
              identifier: 'e3b34f32-8081-4451-83e0-4d5fcfd36512',
              scopedName: 'elr_avap_paimpol'
            },
            {
              operationName: 'DescribeFeatureType',
              identifier: 'e3b34f32-8081-4451-83e0-4d5fcfd36512',
              scopedName: 'elr_avap_paimpol'
            }
          ]
        },
        distributionInfo: {
          transferOptions: [
            {
              onLine: [
                {
                  linkage: 'https://example.com/wfs?SERVICE=WFS&REQUEST=GetCapabilities&',
                  protocol: 'OGC:WFS-2.0.0-http-get-capabilities'
                }
              ]
            }
          ]
        }
      })).toMatchSnapshot()
    })

    it('should map fixtures', () => {
      const fixtures = [
        '1edc94510a82f1e369e27234e1dc674cb85c0f8e'
      ]

      expect.assertions(fixtures.length)

      for (const fixture of fixtures) {
        const record = require(`./__fixtures__/${fixture}`)
        expect(convert(record.content)).toMatchSnapshot()
      }
    })
  })
})
