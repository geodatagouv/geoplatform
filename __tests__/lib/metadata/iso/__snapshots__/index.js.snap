// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.metadata.iso.index convert() extract featureTypes from a service metadata 1`] = `
Object {
  "alternateTitle": undefined,
  "contacts": Array [],
  "description": undefined,
  "featureTypes": Array [
    Object {
      "relatedTo": "7bcc8a8723759135feb7040380a2f2d7c921280f",
      "serviceURL": "https://example.com/wfs",
      "typeName": "bati_avap_paimpol",
    },
    Object {
      "relatedTo": "46a8d50322e881fd4e8b7abc3bc45c38a247532e",
      "serviceURL": "https://example.com/wfs",
      "typeName": "elr_avap_paimpol",
    },
  ],
  "id": undefined,
  "keywords": Array [],
  "metadataType": "ISO 19139",
  "resourceId": undefined,
  "serviceProtocol": "wfs",
  "serviceType": "download",
  "serviceURL": "https://example.com/wfs",
  "title": undefined,
  "type": "service",
}
`;

exports[`lib.metadata.iso.index convert() should map fixtures 1`] = `
Object {
  "alternateTitle": "Base de Données Orthophotographiques",
  "contacts": Array [
    Object {
      "address": Array [
        "IGN - Service Clients",
        "73 avenue de Paris",
      ],
      "country": "FRANCE",
      "email": "sav.bd@ign.fr",
      "organizationName": "Institut National de l'Information Géographique et Forestière (IGN-F)",
      "postalCode": "94165",
      "relatedTo": "metadata",
      "role": "other",
      "town": "SAINT-MANDE cedex",
    },
    Object {
      "address": Array [
        "IGN - Service Clients",
        "73 avenue de Paris",
      ],
      "country": "FRANCE",
      "email": "sav.bd@ign.fr",
      "organizationName": "Institut National de l'Information Géographique et Forestière (IGN-F)",
      "postalCode": "94165",
      "relatedTo": "data",
      "role": "other",
      "town": "SAINT-MANDE cedex",
    },
  ],
  "coveredTerritories": Array [],
  "credit": undefined,
  "description": "Le produit BD ORTHO® V2 est une collection de mosaïques d'orthophotographies numériques en couleurs ou en Infra Rouge couleurs, rectifiées dans la projection adaptée au territoire couvert.",
  "equivalentScaleDenominator": undefined,
  "featureTypes": Array [],
  "id": "IGNF_BDORTHOr_2-0.xml",
  "inspireTheme": Object {
    "id": "oi",
    "label": Object {
      "en": "Orthoimagery",
      "fr": "Ortho-imagerie",
    },
    "uri": "http://inspire.ec.europa.eu/theme/oi",
  },
  "keywords": Array [
    "imageryBaseMapsEarthCover",
    "Ortho-imagerie",
    "ortho-imagerie",
  ],
  "license": null,
  "lineage": "Le nombre de clichés nécessaires pour couvrir un département est important, ces clichés présentent des disparités fortes d'ensoleillement, d'apparence colorée, de paysage, de comportement d'objets (carrières, sables, surfaces d'eau, ...) et parfois de saison. Dans l'idéal, l'ortho-image d'un département doit être continue et homogène sans surface saturée conséquente. En pratique, l'aspect radiométrique de la mosaïque est le résultat d'un compromis sur tout le département tenant compte des disparités mentionnées. Pour la couleur, chaque cliché est corrigé des effets internes d'éclairement et de couleurs liés à l'ensoleillement. L'ensemble des clichés est ensuite homogénéisé et la mosaïque est finalement globalement rehaussée en couleur et en dynamique afin d'obtenir un rendu le plus naturel possible. Pour le canal infra-rouge, chaque cliché est corrigé des effets internes d'éclairement et de couleur liés à l'ensoleillement, puis la mosaïque est corrigée globalement afin d'exploiter au mieux le spectre radiométrique d'une façon proche des canaux rouge et vert qui composent avec le canal infra-rouge les images infra-rouge couleur.
Produit par l'Institut Géographique National (IGN-F)",
  "links": Array [
    Object {
      "description": undefined,
      "href": "http://www.ign.fr/",
      "name": "",
    },
    Object {
      "description": undefined,
      "href": "http://www.geoportail.fr/",
      "name": "",
    },
    Object {
      "description": undefined,
      "href": "http://professionnels.ign.fr/sites/default/files/DL_raster_ORTHO.pdf",
      "name": "DL_raster_ORTHO.pdf",
    },
    Object {
      "description": undefined,
      "href": "http://www.ign.fr/",
      "name": "",
    },
    Object {
      "description": undefined,
      "href": "http://www.geoportail.fr/",
      "name": "",
    },
    Object {
      "description": undefined,
      "href": "http://www.ign.fr/",
      "name": "",
    },
    Object {
      "description": undefined,
      "href": "http://www.geoportail.fr/",
      "name": "",
    },
    Object {
      "description": undefined,
      "href": "http://www.ign.fr/",
      "name": "",
    },
    Object {
      "description": undefined,
      "href": "http://www.geoportail.fr/",
      "name": "",
    },
    Object {
      "description": undefined,
      "href": "http://www.ign.fr/",
      "name": "",
    },
    Object {
      "description": undefined,
      "href": "http://www.geoportail.fr/",
      "name": "",
    },
  ],
  "metadataType": "ISO 19139",
  "purpose": "La BD ORTHO® V2 se compose d'images numériques (sans habillage ni surcharge) et d'indications de géoréférencement. Le contenu informationnel permet localement une utilisation jusqu'au 1 :2,000ième pour mettre en valeur le territoire, enrichir la visualisation des données et des projets.",
  "resourceId": "IGNF_BDORTHOr_2-0",
  "spatialRepresentationType": "grid",
  "spatialResolution": Object {
    "unit": "meter",
    "value": 0.5,
  },
  "status": "onGoing",
  "thumbnails": Array [],
  "title": "BD ORTHO®",
  "topicCategory": "imageryBaseMapsEarthCover",
  "type": "series",
  "updateFrequency": "irregular",
}
`;
